### PR TITLE
ANSI-578 FIPS Compliance Rework

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Updates the code to follow this guide: https://docs.confluent.io/platform/current/security/compliance.html
- `confluent.ssl` role will create pkcs12 keystores and also bcfks when `create_bouncy_castle_keystore` var set to true
- Self signed certs/keys get generated with openssl (instead of keytool). This is because there is a [known issue with openssl v 102 ](https://bugzilla.redhat.com/show_bug.cgi?id=1712023) which is preventing the export of key from keystore file on fips enabled hosts
- No longer uses the bouncy castle jar in the creation of pkcs12 keystores, only when creating bcfks keystores
- kafka will have bcfks and pkcs12 keystores. Needs both because zk tls connection does not support bcfks type
- All other components will continue to have pkcs12 keystores only
- Add `common`, `package`, and `ssl` tags
- Minor fix to c3 properties filters - ANSI-697
- Make path of ca cert customizable - ANSI-698
- Minor bugfixes found during 610 release - ANSI-693
- Added task to handle encrypted private key and unencrypt them. This is necessary for component MTLS health checks

Notes:
- For custom certs, the `ssl_signed_cert_filepath` must be a cert chain starting with root, intermediate(s), then signed cert
- Noticed a running `rngd` service was necessary on my centos-fips vm for keytool commands to complete

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

updated these scenarios to test fips:
- customcerts-rhel
- custombundle-rhel
- mtls-zookeeper TODO review this
- Manually tested on centos ec2 instance with fips enabled w MTLS everywhere
- Manually tested on centos ec2 instance with fips enabled w TLS + sasl plain

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: filter_plugins/filters.py [hunk 1]
```diff
@@ -108,7 +108,8 @@ def split_newline_to_dict(self, string):
                 final_dict[prop_list[0]] = prop_list[1]
         return final_dict
 
-    def listener_properties(self, listeners_dict, default_ssl_enabled, default_pkcs12_enabled, default_ssl_mutual_auth_enabled, default_sasl_protocol,
+    # TODO pkcs12_enabled logic should be reworked
```
Comment (domenicbove) [inline filter_plugins/filters.py:111]: remove this comment

File: filter_plugins/filters.py [hunk 2]
```diff
@@ -125,11 +126,12 @@ def listener_properties(self, listeners_dict, default_ssl_enabled, default_pkcs1
                 final_dict['listener.name.' + listener_name + '.ssl.keystore.password'] = kafka_broker_keystore_storepass
                 final_dict['listener.name.' + listener_name + '.ssl.key.password'] = kafka_broker_keystore_keypass
 
-            if listeners_dict[listener].get('pkcs12_enabled', default_pkcs12_enabled):
+            # if listeners_dict[listener].get('pkcs12_enabled', default_pkcs12_enabled):
```
Comment (domenicbove) [inline filter_plugins/filters.py:129]: remove comment out code

File: roles/confluent.ssl/tasks/create_keystore_and_truststore.yml [hunk 3]
```diff
@@ -11,6 +11,12 @@
   loop:
     - "{{keystore_path}}"
     - "{{truststore_path}}"
+    - "/var/ssl/private/{{service_name}}.keystore.jks"
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/create_keystore_and_truststore.yml:14]: clean this up

File: roles/confluent.ssl/tasks/self_signed_certs.yml [hunk 4]
```diff
@@ -74,19 +82,39 @@
...
       -storetype pkcs12 \
       -alias CARoot \
       -import -file {{ca_cert_path}} \
       -storepass {{keystore_storepass}} \
-      -keypass {{keystore_storepass}} {{extra_args}}
+      -keypass {{keystore_storepass}}
+
+- name: Create Cert Chain
+  shell: |
+    cat {{ca_cert_path}} {{cert_path}} > /var/ssl/private/generation/signed-cert-chain.crt
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/self_signed_certs.yml:94]: might want to put the cert chain into cert_path

File: roles/confluent.ssl/tasks/self_signed_certs.yml [hunk 5]
```diff
@@ -74,19 +82,39 @@
...
+    keytool -noprompt -keystore /var/ssl/private/{{service_name}}.keystore.jks \
       -storetype pkcs12 \
       -alias {{inventory_hostname}} \
-      -import -file {{cert_path}} \
+      -import -file /var/ssl/private/generation/signed-cert-chain.crt \
       -ext "SAN={{hostnames | cert_extension}}" \
       -storepass {{keystore_storepass}} \
-      -keypass {{keystore_storepass}} {{extra_args}}
+      -keypass {{keystore_storepass}}
+
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/self_signed_certs.yml:105]: Missing logic for custom certs option, also investigate provided keystore/truststore

File: roles/confluent.variables/defaults/main.yml [hunk 6]
```diff
@@ -39,7 +39,9 @@ fips_enabled: false
 
 fips_jar_path: /usr/share/java/kafka/bc-fips-*.jar
 fips_provider_class: org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider
-pkcs12_enabled: "{{ true if fips_enabled|bool else false }}"
+
+# TODO this var is dumb, should just use the fips enabled var and remove it
+pkcs12_enabled: "{{fips_enabled}}"
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml:44]: remove all this

File: roles/confluent.variables/defaults/main.yml [hunk 7]
```diff
@@ -294,8 +296,19 @@ kafka_broker_user: "{{kafka_broker_default_user}}"
 ### Only use to customize Linux Group Kafka Service user belongs to. Group must exist on host.
 kafka_broker_group: "{{kafka_broker_default_group}}"
 
-kafka_broker_truststore_path: "/var/ssl/private/kafka_broker.truststore.jks"
-kafka_broker_keystore_path: "/var/ssl/private/kafka_broker.keystore.jks"
+# TODO change jks -> pkcs12
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml:299]: change jks to pkcs12

File: roles/confluent.variables/vars/main.yml [hunk 8]
```diff
@@ -49,13 +49,15 @@ zookeeper_properties:
...
       ssl.quorum.trustStore.password: "{{zookeeper_truststore_storepass}}"
       ssl.clientAuth: "{{ 'need' if zookeeper_ssl_mutual_auth_enabled|bool else 'none' }}"
-  ssl_fips:
-    enabled: "{{ zookeeper_ssl_enabled and pkcs12_enabled }}"
-    properties:
-      ssl.keymanager.algorithm: PKIX
-      ssl.trustmanager.algorithm: PKIX
-      ssl.quorum.keymanager.algorithm: PKIX
-      ssl.quorum.trustmanager.algorithm: PKIX
+  # TODO need to understand why these variables were necessary, its not bc pkcs12 keystores were used bc pkcs12 keystores work without these Variables
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:52]: @prestonmcgowan This PR is a still a work in progress, but the plan is to use type PKCS12 keystores  (created without the bouncy castle jar) for ZK and I've found they do not need these props: `ssl.keymanager.algorithm: PKIX` 

Trying to figure out why they were needed in the first place. Any thoughts?

File: roles/confluent.variables/vars/main.yml [hunk 9]
```diff
@@ -162,23 +164,23 @@ kafka_broker_properties:
...
       zookeeper.clientCnxnSocket: org.apache.zookeeper.ClientCnxnSocketNetty
-      zookeeper.ssl.truststore.location: "{{kafka_broker_truststore_path}}"
+      zookeeper.ssl.truststore.location: "{{kafka_broker_pkcs12_truststore_path}}"
       zookeeper.ssl.truststore.password: "{{kafka_broker_truststore_storepass}}"
-  zk_ssl_fips:
-    enabled: "{{ zookeeper_ssl_enabled and pkcs12_enabled }}"
-    properties:
-      zookeeper.ssl.trustmanager.algorithm: PKIX
-      zookeeper.ssl.truststore.type: PKCS12
+  # zk_ssl_fips:
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:169]: remove commented out code

File: roles/confluent.variables/vars/main.yml [hunk 10]
```diff
@@ -232,19 +235,19 @@ kafka_broker_properties:
...
+      confluent.metadata.server.ssl.keystore.type: BCFKS
+      confluent.metadata.server.ssl.truststore.type: BCFKS
   rbac_external_mds:
     enabled: "{{rbac_enabled and external_mds_enabled}}"
     properties:
       confluent.metadata.bootstrap.servers: "{{mds_broker_bootstrap_servers}}"
   rbac_external_mds_client:
     enabled: "{{rbac_enabled and external_mds_enabled}}"
-    properties: "{{ mds_broker_listener | client_properties(ssl_enabled, pkcs12_enabled, ssl_mutual_auth_enabled, sasl_protocol,
+    properties: "{{ mds_broker_listener | client_properties(ssl_enabled, 'BCFKS', ssl_mutual_auth_enabled, sasl_protocol,
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:250]: this should be a variable its not always set to BCFKS

File: roles/confluent.ssl/tasks/create_keystore_and_truststore.yml (hunk 3 above)
Comment (domenicbove) [reply roles/confluent.ssl/tasks/create_keystore_and_truststore.yml:14]: should think about if we really need to pass keystore paths

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 11]
```diff
@@ -68,6 +71,7 @@
     cert_path: "{{kafka_broker_cert_path}}"
     key_path: "{{kafka_broker_key_path}}"
     export_certs: "{{kafka_broker_export_certs}}"
+    convert_to_bcfks: True
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml:74]: should only be true if fips_enabled

File: roles/confluent.variables/vars/main.yml (hunk 8 above)
Comment (domenicbove) [reply roles/confluent.variables/vars/main.yml:52]: @jeffhuang26 says using PKIX is a good idea and is valid for PKCS12 keystore type, so should leave this code in

File: roles/confluent.ssl/tasks/custom_certs.yml [hunk 12]
```diff
@@ -88,4 +80,6 @@
       -alias CARoot \
       -import -file {{ca_cert_path}} \
       -storepass {{keystore_storepass}} \
-      -keypass {{keystore_storepass}} {{extra_args}}
+      -keypass {{keystore_storepass}}
+
+# TODO add tasks to converge truststore/keystore into BCFKS
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/custom_certs.yml:85]: Need tasks here, should be able to copy from self signed certs file

Comment (domenicbove) [reply roles/confluent.ssl/tasks/custom_certs.yml:85]: need test as well

File: roles/confluent.test/molecule/zookeeper-mtls/molecule.yml [hunk 13]
```diff
@@ -86,6 +86,54 @@ platforms:
     privileged: true
     networks:
       - name: confluent
+  - name: kafka-connect1
+    hostname: kafka-connect1.confluent
+    groups:
+      - kafka_connect
+    image: geerlingguy/docker-centos7-ansible
+    dockerfile: ../Dockerfile.j2
```
Comment (domenicbove) [inline roles/confluent.test/molecule/zookeeper-mtls/molecule.yml:94]: create new fips dockerfile: https://github.com/arhea/docker-fips-library/blob/master/centos/Dockerfile

File: roles/confluent.variables/vars/main.yml [hunk 14]
```diff
@@ -703,13 +706,13 @@ ksql_properties:
...
     properties:
       ssl.client.auth: 'true'
-  pkcs12:
-    enabled: "{{ pkcs12_enabled|bool }}"
-    properties:
-      ssl.keymanager.algorithm: PKIX
-      ssl.trustmanager.algorithm: PKIX
-      ssl.keystore.type: PKCS12
-      ssl.truststore.type: PKCS12
+  # pkcs12:
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:709]: remove this

File: roles/confluent.ssl/tasks/self_signed_certs.yml (hunk 4 above)
Comment (domenicbove) [reply roles/confluent.ssl/tasks/self_signed_certs.yml:94]: doesnt work w the health checks leaving as is

File: roles/confluent.variables/defaults/main.yml [hunk 15]
```diff
@@ -199,8 +195,8 @@ zookeeper_client_port: "{{'2182' if zookeeper_ssl_enabled|bool else '2181'}}"
 ### SASL Mechanism for Zookeeper Server to Server and Server to Client Authentication. Options are none, kerberos, digest. Server to server auth only working for digest-md5
 zookeeper_sasl_protocol: "{{sasl_protocol if sasl_protocol == 'kerberos' else 'none'}}"
 
-zookeeper_truststore_path: "/var/ssl/private/zookeeper.truststore.jks"
-zookeeper_keystore_path: "/var/ssl/private/zookeeper.keystore.jks"
+zookeeper_truststore_path: "/var/ssl/private/zookeeper.truststore.p12"
+zookeeper_keystore_path: "/var/ssl/private/zookeeper.keystore.p12"
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml:199]: this change is problematic... found that when jks files are named with p12 kafka/zookeeper fails to load them in
mostly problematic with provided keystore/truststore

anyhow this change isnt necessary to get fips to work, can be saved for later work

File: roles/confluent.ssl/tasks/custom_certs.yml [hunk 16]
```diff
@@ -46,29 +38,51 @@
...
+      -in /var/ssl/private/generation/signed-cert-chain.crt \
       -inkey {{key_path}} \
+      {% if ssl_key_password is defined %}-passin pass:{{ssl_key_password}}{% endif %} \
       -out /var/ssl/private/generation/client.p12 \
       -name kafkassl \
       -passout pass:mykeypassword
-  when: ssl_key_password is not defined
+  # when: ssl_key_password is defined
+
+# - name: Put Key and Signed Cert into pkcs12 Format without Key Password
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/custom_certs.yml:77]: remove commented out code

File: roles/confluent.ssl/tasks/custom_certs.yml [hunk 17]
```diff
@@ -88,4 +102,20 @@
...
+      -srcalias kafkassl \
+      -srcstoretype PKCS12 \
+      -destkeystore {{bcfks_keystore_path}} \
+      -deststorepass {{keystore_storepass}} \
+      -destalias {{inventory_hostname}}  \
+      -deststoretype BCFKS \
+      -providername BCFIPS \
+      -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+      -providerpath /usr/share/java/kafka/bc-fips-*.jar
+  when: convert_to_bcfks|default(False)|bool
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/custom_certs.yml:121]: put the default into the role defaults

File: roles/confluent.ssl/tasks/export_certs_from_keystore_and_truststore-back.yml [hunk 18]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/export_certs_from_keystore_and_truststore-back.yml:1]: put this back

File: roles/confluent.ssl/tasks/main.yml [hunk 19]
```diff
@@ -34,3 +34,14 @@
...
+- name: Set BCFKS Truststore and Keystore File Permissions
+  file:
+    path: "{{item}}"
+    owner: "{{user}}"
+    group: "{{group}}"
+    mode: 0640
+  loop:
+    - "{{bcfks_keystore_path}}"
+    - "{{bcfks_truststore_path}}"
+  when: convert_to_bcfks|default(False)|bool
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/main.yml:47]:
```suggestion
  when: convert_to_bcfks|bool
```

File: roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml [hunk 20]
```diff
@@ -11,4 +11,4 @@
 
 - name: Export Certs from Keystore and Truststore
   include_tasks: export_certs_from_keystore_and_truststore.yml
-  when: export_certs|bool
+  when: export_certs|bool or convert_to_bcfks|default(False)|bool
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml:14]: revert this, dont do anything for provided keystores/fips

File: roles/confluent.ssl/tasks/self_signed_certs.yml [hunk 21]
```diff
@@ -9,22 +9,30 @@
...
+      -srcalias CARoot \
+      -srcstoretype PKCS12 \
+      -destkeystore {{bcfks_truststore_path}} \
+      -deststorepass {{truststore_storepass}} \
+      -destalias CARoot \
+      -deststoretype BCFKS \
+      -providername BCFIPS \
+      -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+      -providerpath /usr/share/java/kafka/bc-fips-*.jar
+  when: convert_to_bcfks|default(False)|bool
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/self_signed_certs.yml:35]:
```suggestion
  when: convert_to_bcfks|bool
```

File: roles/confluent.ssl/tasks/self_signed_certs.yml [hunk 22]
```diff
@@ -79,14 +87,34 @@
...
+      -srcalias {{inventory_hostname}} \
+      -srcstoretype PKCS12 \
+      -destkeystore {{bcfks_keystore_path}} \
+      -deststorepass {{keystore_storepass}} \
+      -destalias {{inventory_hostname}}  \
+      -deststoretype BCFKS \
+      -providername BCFIPS \
+      -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+      -providerpath /usr/share/java/kafka/bc-fips-*.jar
+  when: convert_to_bcfks|default(False)|bool
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/self_signed_certs.yml:120]:
```suggestion
  when: convert_to_bcfks|bool
```

File: roles/confluent.test/molecule/mtls-custombundle-rhel/verify.yml [hunk 23]
```diff
@@ -48,7 +30,7 @@
       vars:
         file_path: /etc/schema-registry/schema-registry.properties
         property: ssl.keystore.location
-        expected_value: /var/ssl/private/schema_registry.keystore.jks
+        expected_value: /var/ssl/private/schema_registry.keystore.p12
```
Comment (domenicbove) [inline roles/confluent.test/molecule/mtls-custombundle-rhel/verify.yml:33]:
```suggestion
        expected_value: /var/ssl/private/schema_registry.keystore.jks
```

File: roles/confluent.test/molecule/mtls-custombundle-rhel/verify.yml [hunk 24]
```diff
@@ -69,7 +51,7 @@
       vars:
         file_path: /etc/kafka-rest/kafka-rest.properties
         property: ssl.keystore.location
-        expected_value: /var/ssl/private/kafka_rest.keystore.jks
+        expected_value: /var/ssl/private/kafka_rest.keystore.p12
```
Comment (domenicbove) [inline roles/confluent.test/molecule/mtls-custombundle-rhel/verify.yml:54]:
```suggestion
        expected_value: /var/ssl/private/kafka_rest.keystore.jks
```

File: roles/confluent.test/molecule/mtls-custombundle-rhel/verify.yml [hunk 25]
```diff
@@ -81,7 +63,7 @@
       vars:
         file_path: /etc/kafka/connect-distributed.properties
         property: ssl.keystore.location
-        expected_value: /var/ssl/private/kafka_connect.keystore.jks
+        expected_value: /var/ssl/private/kafka_connect.keystore.p12
```
Comment (domenicbove) [inline roles/confluent.test/molecule/mtls-custombundle-rhel/verify.yml:66]:
```suggestion
        expected_value: /var/ssl/private/kafka_connect.keystore.jks
```

File: roles/confluent.test/molecule/mtls-provided-fips-ubuntu/certs-create.sh [hunk 26]
```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
```
Comment (domenicbove) [inline roles/confluent.test/molecule/mtls-provided-fips-ubuntu/certs-create.sh:1]: remove this scenario

File: roles/confluent.test/molecule/zookeeper-mtls/molecule.yml (hunk 13 above)
Comment (domenicbove) [reply roles/confluent.test/molecule/zookeeper-mtls/molecule.yml:94]: This did not work 
```
% docker run --rm -it fips:latest cat /proc/sys/crypto/fips_enabled
cat: /proc/sys/crypto/fips_enabled: No such file or directory
```

File: VARIABLES.md [hunk 27]
```diff
@@ -78,7 +78,7 @@ Default:  /opt/prometheus/jmx_prometheus_javaagent.jar
 
 ### fips_enabled
 
-Boolean to have cp-ansible configure components with FIPS security settings
+Boolean to have cp-ansible configure components with FIPS security settings. Must have ssl_enabled: true and use Java 8
```
Comment (domenicbove) [inline VARIABLES.md:81]: lets test on java 11

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 28]
```diff
@@ -68,13 +71,17 @@
...
+    bcfks_truststore_path: "{{kafka_broker_bcfks_truststore_path}}"
+    bcfks_keystore_path: "{{kafka_broker_bcfks_keystore_path}}"
   when: >
     zookeeper_ssl_enabled|bool or
     kafka_broker_listeners | ssl_required(ssl_enabled) or
     kafka_broker_rest_ssl_enabled|bool or
     mds_broker_listener.ssl_enabled|bool or
     mds_tls_enabled|bool or
     schema_registry_ssl_enabled|bool
+  tags: ssl
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml:84]: add this tag elsewhere

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 29]
```diff
@@ -68,13 +71,17 @@
     cert_path: "{{kafka_broker_cert_path}}"
     key_path: "{{kafka_broker_key_path}}"
     export_certs: "{{kafka_broker_export_certs}}"
+    convert_to_bcfks: "{{fips_enabled}}"
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml:74]: rename variable to `create_bcfks`

File: roles/confluent.ssl/tasks/main.yml [hunk 30]
```diff
@@ -22,6 +22,7 @@
   loop:
     - "{{keystore_path}}"
     - "{{truststore_path}}"
+  failed_when: false
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/main.yml:25]: remove this

File: roles/confluent.test/molecule/zookeeper-mtls/molecule.yml (hunk 13 above)
Comment (domenicbove) [reply roles/confluent.test/molecule/zookeeper-mtls/molecule.yml:94]: testing in docker containers did not work, maybe add an automated test on aws

File: roles/confluent.ssl/tasks/self_signed_certs.yml [hunk 31]
```diff
@@ -1,60 +1,23 @@
...
-    keytool -keystore {{keystore_path}} \
-      -storetype pkcs12 \
-      -alias {{inventory_hostname}} \
-      -certreq -file /var/ssl/private/generation/client.csr \
-      -ext "SAN={{hostnames | cert_extension}}" \
-      -storepass {{keystore_storepass}} \
-      -keypass {{keystore_storepass}} {{extra_args}}
+    openssl req -new -key {{key_path}} \
+      -out /var/ssl/private/generation/client.csr \
+      -subj '/CN=kafka_broker/OU=TEST/O=CONFLUENT/L=PaloAlto/ST=Ca/C=US'
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/self_signed_certs.yml:20]: shouldnt be kafka_broker here

File: fips.yml [hunk 32]
```diff
@@ -0,0 +1,78 @@
+---
```
Comment (domenicbove) [inline fips.yml:1]: probably dont include this

File: roles/confluent.ssl/tasks/create_keystores_from_certs.yml [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+
+# Not working w ca bundle in the ca.crt, does not seem to matter though
+# - name: Create Temporary PKCS12 File with Key and Signed Cert Chain
+#   shell: |
+#     openssl pkcs12 -export -chain \
+#       -in /var/ssl/private/generation/signed-cert-chain.crt \
+#       -inkey {{key_path}} \
+#       {% if ssl_key_password is defined %}-passin pass:{{ssl_key_password}}{% endif %} \
+#       -out /var/ssl/private/generation/client.p12 \
+#       -name localhost \
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/create_keystores_from_certs.yml:52]: remove commented out code

File: roles/confluent.ssl/tasks/create_keystores_from_certs.yml [hunk 34]
```diff
@@ -0,0 +1,137 @@
...
+#       -passout pass:mykeypassword
+#   register: output
+
+- name: Create Temporary PKCS12 File with Key and Signed Cert Chain
+  shell: |
+    openssl pkcs12 -export \
+      -in /var/ssl/private/{{service_name}}.chain \
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/create_keystores_from_certs.yml:72]: question is, is there value of keeping this chain around

File: roles/confluent.ssl/tasks/custom_certs.yml [hunk 35]
```diff
@@ -5,87 +5,27 @@
     src: "{{ssl_ca_cert_filepath}}"
     dest: "{{ca_cert_path}}"
 
+# - name: Copy Signed Cert to Host
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/custom_certs.yml:8]: remove commented out code

File: roles/confluent.ssl/tasks/self_signed_certs.yml [hunk 36]
```diff
@@ -1,66 +1,40 @@
...
+    openssl req -new -key {{key_path}} \
+      -out /var/ssl/private/generation/client.csr \
+      -subj '/CN={{service_name}}/OU=TEST/O=CONFLUENT/L=PaloAlto/ST=Ca/C=US'
 
 - name: Create Openssl Config for SAN
   template:
     src: openssl-san.cnf.j2
     dest: /var/ssl/private/generation/openssl-san.cnf
 
+# - name: Sign CSR with the Certificate Authority
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/self_signed_certs.yml:27]: remove commented out code

File: roles/confluent.variables/vars/main.yml [hunk 37]
```diff
@@ -145,6 +138,16 @@ kafka_broker_properties:
...
+      ssl.keymanager.algorithm: PKIX
+      ssl.trustmanager.algorithm: PKIX
+      ssl.keystore.type: BCFKS
+      ssl.truststore.type: BCFKS
+      ssl.truststore.location: "{{kafka_broker_truststore_path}}"
+      ssl.truststore.password: "{{kafka_broker_truststore_storepass}}"
+      ssl.keystore.location: "{{kafka_broker_keystore_path}}"
+      ssl.keystore.password: "{{kafka_broker_keystore_storepass}}"
+      ssl.key.password: "{{kafka_broker_keystore_storepass}}"
+      ssl.enabled.protocol: TLSv1.2
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:150]: do you need the ssl.enabled.protocol: TLSv1.2 added to each listener? or is here ok?

are any of these props needed if they are added to the listeners

File: roles/confluent.variables/vars/main.yml [hunk 38]
```diff
@@ -188,20 +181,20 @@ kafka_broker_properties:
     properties:
       confluent.schema.registry.url: "{{schema_registry_url}}"
   sr_ssl:
-    enabled: "{{ 'schema_registry' in groups and schema_registry_ssl_enabled|bool }}"
+    enabled: "{{ 'schema_registry' in groups and schema_registry_ssl_enabled }}"
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:184]: add schema validation logic, its missing

File: roles/confluent.variables/vars/main.yml [hunk 39]
```diff
@@ -188,20 +181,20 @@ kafka_broker_properties:
...
       confluent.ssl.keystore.password: "{{kafka_broker_keystore_storepass}}"
       confluent.ssl.key.password: "{{kafka_broker_keystore_keypass}}"
-  sr_ssl_fips:
-    enabled: "{{ 'schema_registry' in groups and schema_registry_ssl_enabled|bool and pkcs12_enabled}}"
-    properties:
-      confluent.schema.registry.ssl.keymanager.algorithm: PKIX
-      confluent.schema.registry.ssl.trustmanager.algorithm: PKIX
-      confluent.schema.registry.ssl.keystore.type: PKCS12
-      confluent.schema.registry.ssl.truststore.type: PKCS12
+  # sr_ssl_fips:
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:191]: can sr connection use the bcfks?

File: VARIABLES.md (hunk 27 above)
Comment (domenicbove) [reply VARIABLES.md:81]: confirmed not working on java 11

Comment (domenicbove) [inline VARIABLES.md:81]: Does not work with provided keystore truststore

**Review by xnulinu — APPROVED** [review]

**Pull request merged**